src/shared/api/CivicAPI.ts

```diff
@@ -0,0 +1,265 @@
+        return getVariant(variantId, variantName, geneId)
+        .then(function(result: ICivicGeneVariant) {
+            if (result) {
+                if (!variantMap[geneSymbol]) {
+                    variantMap[geneSymbol] = {};
+                }
+                variantMap[geneSymbol][variantName] = result;
+            }
+        })
+    }
+    
+    let service = {
+        
+        /**
+         * Asynchronously return a map with Civic information from the genes given.
+         */
+        getCivicGenes: function(geneSymbols: Array<string>): JQueryPromise<ICivicData> {
+
+            // Assemble a list of promises, each of which will retrieve a batch of genes
+            let promises: Array<JQueryPromise<void>> = [];
+            let ids = '';
+            geneSymbols.forEach(function(geneSymbol: string) {
+                // Check if we already have it in the cache
+                if (civicData.hasOwnProperty(geneSymbol)) {
+                    return;
+                }
+
+                // Add the symbol to the list
+                if (ids.length > 0) {
+                    ids += ',';
+                }
+                ids += geneSymbol;
+
+                // To prevent the request from growing too large, we send it off
+                // when it reaches this limit and start a new one
+                if (ids.length >= 1900) {
+                    promises.push(setCivicGenesBatch(ids));
+                    ids = '';
+                }
+            });
+            if (ids.length > 0) {
+                promises.push(setCivicGenesBatch(ids));
+            }
+
+            // We're explicitly waiting for all promises to finish (done or fail).
+            // We are wrapping them in another promise separately, to make sure we also 
+            // wait in case one of the promises fails and the other is still busy.
+            let wrappedPromises = $.map(promises, function(promise: JQueryPromise<void>) {
+                let wrappingDeferred = $.Deferred();
+                promise.always(function (result) {
+                    wrappingDeferred.resolve();
+                });
+                return wrappingDeferred.promise();
+            });
+            
+            let mainPromise = $.when.apply($, wrappedPromises);
+
+            return mainPromise.then(function() {
+                return civicData;
+            });
+        },
+        
+        /**
+         * Asynchronously retrieve a map with Civic information from the mutationSpecs given for all genes in civicData.
+         * If no mutationSpecs are given, then return the Civic information of all the CNA variants of the genes in civicData.
+         */
+        getCivicVariants: function(civicData: ICivicData, mutationSpecs?: Array<{gene:{hugoGeneSymbol: string}, proteinChange: string}>): JQueryPromise<ICivicVariant> {
```

Comment:
` Array<{gene:{hugoGeneSymbol: string}, proteinChange: string}>` is a bit confusing. We are actually passing Mutation instances to this function. How about something like this:
```JavaScript
function getCivicVariants<MutationSpec extends {gene:{hugoGeneSymbol: string}, proteinChange: string}>(..., mutationSpecs?: MutationSpecs[], ...)
```

That would allow passing a `Mutation` instance to this function in a clearer way.